### ISSUE TITLE
circular dependency (invalid esm) in v5 at `/dist/es/Seq.js` and `dist/es/Collection.js`

### ISSUE THREAD

**Issue by iambumblehead** [immutable-js/immutable-js#2019]
[Collection.js](https://unpkg.com/browse/immutable@5.0.0-rc.1/dist/es/Collection.js#L25) imports Seq.js and [Seq.js](https://unpkg.com/browse/immutable@5.0.0-rc.1/dist/es/Seq.js#L26) imports Collection.js

Because circular dependencies aren't supported by esm, the files aren't usable as-is and must be transpiled.

Also, these lines [in Seq.js look strange.](https://unpkg.com/browse/immutable@5.0.0-rc.1/dist/es/Seq.js#L48)
```javascript
if ( Collection ) Seq.__proto__ = Collection;
Seq.prototype = Object.create( Collection && Collection.prototype );
Seq.prototype.constructor = Seq;
```
When Collection is imported through a circular dependency, its value is falsey `undefined` which causes a runtime error when it is passed to `Object.create`

related https://github.com/immutable-js/immutable-js/issues/1961

**Comment by iambumblehead** [comment]
building immutable with this patch at the [5.x branch](https://github.com/immutable-js/immutable-js/pull/2011) gives me an es module that is working here
```diff
diff --git a/resources/rollup-config.mjs b/resources/rollup-config.mjs
index bf32fc38..99c070b8 100644
--- a/resources/rollup-config.mjs
+++ b/resources/rollup-config.mjs
@@ -36,10 +36,8 @@ export default [
       {
         banner: copyright,
         name: 'Immutable',
-        dir: path.join(DIST_DIR, 'es'),
+        file: path.join(DIST_DIR, 'immutable.es.js'),
         format: 'es',
-        preserveModules: true,
-        preserveModulesRoot: SRC_DIR,
         sourcemap: false,
       },
     ],
```

**Comment by jdeniau** [comment]
@kyle-leonhard do,you have a reproducible exemple? Can you provide your nodejs version and / or your bundler?

**Comment by jdeniau** [comment]
@kyle-leonhard this should be resolved in 5.0.1 thanks to @iambumblehead
